cluster-tests/tests/filesystem/test_002_glusterfs_filesystem_plugin.py

```diff
@@ -0,0 +1,275 @@
+
+
+def test_003_read_write_file(request):
+    depends(request, ['HAS_ROOT_UUID'])
+    STRING_DATA_SAMPLE = 'FLUFFY_BUNNIES'
+    ip = CLUSTER_IPS[0]
+
+    with create_test_object(ip, False, **{
+        'volume_name': CLUSTER_INFO['GLUSTER_VOLUME'],
+        'path': 'GFAPI_TESTFILE_RW',
+    }) as gl_obj:
+        assert gl_obj['file_type']['parsed'] == 'FILE'
+
+        # First start with writing / reading via string
+        res = make_ws_request(ip, {
+            'msg': 'method',
+            'method': 'gluster.filesystem.pwrite',
+            'params': [{
+                'volume_name': CLUSTER_INFO['GLUSTER_VOLUME'],
+                'uuid': gl_obj['uuid'],
+                'payload': STRING_DATA_SAMPLE
+            }]
+        })
+
+        assert res.get('error') is None, res
+
+        res = make_ws_request(ip, {
+            'msg': 'method',
+            'method': 'gluster.filesystem.contents',
+            'params': [{
+                'volume_name': CLUSTER_INFO['GLUSTER_VOLUME'],
+                'uuid': gl_obj['uuid']
+            }]
+        })
+
+        assert res.get('error') is None, res
+        assert res['result'] == STRING_DATA_SAMPLE
+
+        # Now write binary content
+        # new pwrite() request starts at offset 0 and so
+        # this will overwrite the previous contents
+        res = make_ws_request(ip, {
+            'msg': 'method',
+            'method': 'gluster.filesystem.pwrite',
+            'params': [{
+                'volume_name': CLUSTER_INFO['GLUSTER_VOLUME'],
+                'uuid': gl_obj['uuid'],
+                'payload': BINARY_DATA_SAMPLE,
+                'payload_type': 'BINARY'
+            }]
+        })
+        assert res.get('error') is None, res
+
+        res = make_ws_request(ip, {
+            'msg': 'method',
+            'method': 'gluster.filesystem.pread',
+            'params': [{
+                'volume_name': CLUSTER_INFO['GLUSTER_VOLUME'],
+                'uuid': gl_obj['uuid'],
+                'options': {'offset': 0, 'cnt': len(BINARY_DATA_SAMPLE)}
+            }]
+        })
+        assert res.get('error') is None, res
+        assert res['result'] == BINARY_DATA_SAMPLE
+
+        res = make_ws_request(ip, {
+            'msg': 'method',
+            'method': 'gluster.filesystem.contents',
+            'params': [{
+                'volume_name': CLUSTER_INFO['GLUSTER_VOLUME'],
+                'uuid': gl_obj['uuid'],
+                'options': {'file_output_type': 'BINARY'}
+            }]
+        })
+        assert res.get('error') is None, res
+        assert res['result'] == BINARY_DATA_SAMPLE
+
+
+def test_004_setattrs(request):
+    depends(request, ['HAS_ROOT_UUID'])
+    ip = CLUSTER_IPS[0]
+
+
+    with create_test_object(ip, True, **{
+        'volume_name': CLUSTER_INFO['GLUSTER_VOLUME'],
+        'path': 'GFAPI_TESTDIR_SETATTRS',
+    }) as gl_obj:
+        # create a bunch of test files
+        for i in range(0, 100):
```

Comment:
Is this a magic number? Do we need that many for some special reason? Just thinking of 200+ calls for this single test

Reply:
Magic. I can reduce to something smaller if you wish.

Reply:
I just thought it would generate too many calls for no good reason. But if you think thats fine, so be it